core/unit_test/CMakeLists.txt

```diff
@@ -338,6 +339,35 @@ KOKKOS_ADD_ADVANCED_TEST( UnitTest_PushFinalizeHook_terminate
     ALWAYS_FAIL_ON_ZERO_RETURN
 )
 
+if(Kokkos_ENABLE_PROFILING)
+  add_library(
+     printer-tool SHARED
+     tools/printing-tool.cpp
+  )
+
+  KOKKOS_ADD_TEST_EXECUTABLE(
+    ProfilingAllCalls
+    tools/TestAllCalls.cpp
+  )
+
+  KOKKOS_ADD_TEST( NAME ProfilingTestParallelFor
+    EXE  ProfilingAllCalls
+    TOOL ${CMAKE_CURRENT_BINARY_DIR}/libprinter-tool.so
```

Comment:
I both wouldn't assume that directory or suffix of the library (although 99% of the time what you have is probably correct).

````
get_target_property(PRINTER_LIB printer-tool LOCATION)
KOKKOS_ADD_TEST( NAME ProfilingTestParallelFor
    EXE  ProfilingAllCalls
    TOOL ${PRINTER_LIB}
...

Reply:
Coool, this actually solves my first bug (this test crashing on Mac where it's called dylib rather than so)

Reply:
Your comment below is what I wound up using instead of this